mlog/log_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+package mlog_test
+
+import (
+	"context"
+	"sync"
+	"testing"
+
+	"github.com/mattermost/mattermost-server/v5/mlog"
+	"github.com/stretchr/testify/require"
+)
+
+// Test race condition when shutting down advanced logging. This test must run with the -race flag in order to verify
+// that there is no race.
+func TestLogger_ShutdownAdvancedLoggingRace(t *testing.T) {
+	logger := mlog.NewLogger(&mlog.LoggerConfiguration{
+		EnableConsole: true,
+		ConsoleJson:   true,
+		EnableFile:    false,
+		FileLevel:     mlog.LevelInfo,
+	})
+	started := make(chan bool)
+	ctx, cancel := context.WithCancel(context.Background())
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		started <- true
+
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			default:
+				logger.Debug("testing...")
+			}
+		}
+	}()
+
+	require.True(t, <-started, "goroutine failed to start")
```

Comment:
Is the `require` check really needed? It's hard to see this ever failing :D

Reply:
Fair :D.